tests/subsys/usb/bos/src/test_bos.c

```diff
@@ -203,15 +203,16 @@ static void test_usb_bos_macros(void)
 
 	/* usb_bos_fix_total_length(); corrected with register */
 
-	hexdump((void *)hdr, len);
-	hexdump((void *)&webusb_bos_descriptor, len);
-	hexdump((void *)&webusb_bos_descriptor_2, len);
-
 	zassert_true(len ==
 		     sizeof(struct usb_bos_descriptor) +
 		     sizeof(cap_webusb) +
 		     sizeof(cap_msosv2),
 		     "Incorrect calculated length");
+
+	hexdump((void *)hdr, len);
+	hexdump((void *)&webusb_bos_descriptor, sizeof(cap_webusb));
+	hexdump((void *)&webusb_bos_descriptor_2, sizeof(cap_msosv2));
```

Comment:
Please just move one hexdump, why do you need 3?

Reply:
is it not better to keep the order, first dump all data and then dump extracted data

Reply:
Sorry, I did not express my self well, I think `hexdump((void *)hdr, len);` should not be moved because It would be beneficial to see it even if the test (L:210) fails.

Reply:
@jfischer-phytec-iot no problem, moved the block back

Reply:
If `len` is not correct we may access wrong memory, etc. If there is a check that dump should always come after the check.

Reply:
len is  `__usb_bos_desc_end - __usb_bos_desc_start`, which are pointer to descriptor section, and even if they are wrong I want to see how memory looks there, otherwise one do not need hexdump at all.